includes/class-paybutton-ajax.php

```
@@ -222,40 +297,63 @@ public function mark_payment_successful() {
         $tx_timestamp = isset( $_POST['tx_timestamp'] ) ? sanitize_text_field( $_POST['tx_timestamp'] ) : '';
         // NEW: Address passed from front-end if user is not logged in
         $user_address = isset( $_POST['user_address'] ) ? sanitize_text_field( $_POST['user_address'] ) : '';
+        $unlock_token  = isset( $_POST['unlock_token'] ) ? sanitize_text_field( $_POST['unlock_token'] ) : '';
+
+        if ( $post_id <= 0 || empty( $tx_hash ) || empty( $user_address ) || empty( $unlock_token ) ) {
+                wp_send_json_error( array( 'message' => 'Missing required payment fields.' ), 400 );
+        }
 
         $mysql_timestamp = '0000-00-00 00:00:00';
```

PHPMD (2.15.0)

Avoid unused local variables such as '$mysql_timestamp'. (undefined)

(UnusedLocalVariable)